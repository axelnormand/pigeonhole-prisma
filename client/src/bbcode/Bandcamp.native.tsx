import React from 'react';
import { WebView } from 'react-native-webview';

export const Bandcamp: React.FC<{ album?: string; track?: string }> = ({
  album,
  track,
}) => {
  if (album && track) {
    const url = `https://bandcamp.com/EmbeddedPlayer/album=${album}/size=large/artwork=small/bgcol=ffffff/linkcol=0687f5/tracklist=false/track=${track}`;
    return (
      <WebView
        testID="bbcode-bandcamp"
        style={{ width: 310, height: 500 }}
        source={{
          uri: url,
        }}
      />
    );
  }

  if (album) {
    const url = `https://bandcamp.com/EmbeddedPlayer/album=${album}/size=large/bgcol=ffffff/linkcol=0687f5/tracklist=true`;
    return (
      <WebView
        testID="bbcode-bandcamp"
        style={{ width: 310, height: 500 }}
        source={{
          uri: url,
        }}
      />
    );
  }

  // just track
  const url = `https://bandcamp.com/EmbeddedPlayer/track=${track}/size=large/bgcol=ffffff/linkcol=0687f5/tracklist=false`;
  return (
    <WebView
      testID="bbcode-bandcamp"
      style={{ width: 310, height: 500 }}
      source={{
        uri: url,
      }}
    />
  );
};
