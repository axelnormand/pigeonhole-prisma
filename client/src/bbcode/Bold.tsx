import React from 'react';
import { Text } from '@ui-kitten/components';

export const Bold: React.FC<{ children: string }> = ({ children }) => (
  <Text style={{ fontWeight: 'bold' }} testID="bbcode-bold">
    {children}
  </Text>
);
