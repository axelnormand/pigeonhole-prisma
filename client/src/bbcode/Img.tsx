import React from 'react';
import Image from 'react-native-scalable-image';
import { Dimensions } from 'react-native';

export const Img: React.FC<{ url: string }> = ({ url }) => (
  <Image
    width={Dimensions.get('window').width} // height will be calculated automatically
    source={{ uri: url }}
  />
);
