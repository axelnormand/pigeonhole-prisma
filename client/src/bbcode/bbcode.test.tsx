import React from 'react';
import { Linking } from 'react-native';
import {
  render as testRender,
  getNodeText,
  fireEvent,
} from '@testing-library/react-native';
import { parse } from './bbcode';
import { ApplicationProvider } from '@ui-kitten/components';
import { mapping, dark } from '@eva-design/eva';

const openURLSpy = jest.spyOn(Linking, 'openURL');

const render = (el: React.ReactNode) =>
  testRender(
    <ApplicationProvider mapping={mapping} theme={dark}>
      {el}
    </ApplicationProvider>,
  );

it('works with no bbcode', () => {
  const text = 'Hello dude';
  expect(parse(text)).toEqual(text);
});

it('works with bold bbcode', () => {
  const text = 'Hello [b]dude[/b] wassup';
  const { getByTestId } = render(parse(text));
  expect(getNodeText(getByTestId('bbcode-bold'))).toEqual('dude');
});

it('works with url bbcode', () => {
  const url = 'https://dude.com';
  const text = `Hello [url]${url}[/url] wassup`;
  const { getByTestId } = render(parse(text));

  const urlComp = getByTestId('bbcode-url');
  expect(getNodeText(urlComp)).toEqual(url);

  fireEvent.press(urlComp);
  expect(openURLSpy).toBeCalledWith(url);
});

it('works with url and link text bbcode', () => {
  const linkText = `DUDE`;
  const url = 'https://dude.com';
  const text = `Hello [url=https://dude.com]${linkText}[/url] wassup`;
  const { getByTestId } = render(parse(text));

  const urlComp = getByTestId('bbcode-url');
  expect(getNodeText(urlComp)).toEqual(linkText);

  fireEvent.press(urlComp);
  expect(openURLSpy).toBeCalledWith(url);
});

it('works with youtube bbcode', () => {
  const videoId = '123';
  const text = `Hello [youtube]https://youtube.com?v=${videoId}[/youtube] wassup`;
  const { getByTestId } = render(parse(text));

  const youtubeComp = getByTestId('bbcode-youtube');
  expect(youtubeComp.getProp('source')).toEqual({
    uri: `https://www.youtube.com/embed/${videoId}?playsinline=1&fs=1`,
  });
});

it('works with 2 bbcodes', () => {
  const text = '[i]Hello[/i] [b]dude[/b] wassup';
  const { getByTestId } = render(parse(text));

  expect(getNodeText(getByTestId('bbcode-italic'))).toEqual('Hello');
  expect(getNodeText(getByTestId('bbcode-bold'))).toEqual('dude');
});
