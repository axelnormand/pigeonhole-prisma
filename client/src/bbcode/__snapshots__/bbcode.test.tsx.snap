// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works 2 bbcodes 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          Object {
            "fontStyle": "italic",
          },
        ]
      }
      testID="bbcode-italic"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
      testID="bbcode-bold"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      dude
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
       wassup
    </Text>
  </View>
</View>
`;

exports[`works with bold bbcode 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello 
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          Object {
            "fontWeight": "bold",
          },
        ]
      }
      testID="bbcode-bold"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      dude
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
       wassup
    </Text>
  </View>
</View>
`;

exports[`works with no bbcode 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello dude
    </Text>
  </View>
</View>
`;

exports[`works with quote bbcode and more text 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "background-basic-color-1",
          },
          Array [
            Object {
              "borderRadius": 8,
              "borderWidth": 2,
              "padding": 16,
            },
            Object {
              "borderColor": "#C5CEE0",
            },
          ],
        ]
      }
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            Object {
              "fontStyle": "italic",
            },
          ]
        }
        testID="bbcode-quote-name"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        Foo
         wrote:
      </Text>
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            undefined,
          ]
        }
        testID="bbcode-quote-text"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              undefined,
            ]
          }
          testID="bbcode-plain"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
          hello, world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`works with quote bbcode and name 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "background-basic-color-1",
          },
          Array [
            Object {
              "borderRadius": 8,
              "borderWidth": 2,
              "padding": 16,
            },
            Object {
              "borderColor": "#C5CEE0",
            },
          ],
        ]
      }
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            Object {
              "fontStyle": "italic",
            },
          ]
        }
        testID="bbcode-quote-name"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        Foo
         wrote:
      </Text>
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            undefined,
          ]
        }
        testID="bbcode-quote-text"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              undefined,
            ]
          }
          testID="bbcode-plain"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
          hello, world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`works with quote bbcode and nested text bbcodes 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "background-basic-color-1",
          },
          Array [
            Object {
              "borderRadius": 8,
              "borderWidth": 2,
              "padding": 16,
            },
            Object {
              "borderColor": "#C5CEE0",
            },
          ],
        ]
      }
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            Object {
              "fontStyle": "italic",
            },
          ]
        }
        testID="bbcode-quote-name"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        Foo
         wrote:
      </Text>
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            undefined,
          ]
        }
        testID="bbcode-quote-text"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "fontStyle": "italic",
              },
            ]
          }
          testID="bbcode-italic"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
          Hello
        </Text>
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              Object {
                "fontWeight": "bold",
              },
            ]
          }
          testID="bbcode-bold"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
          dude
        </Text>
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              undefined,
            ]
          }
          testID="bbcode-plain"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
           wassup
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`works with quote bbcode and no name 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "background-basic-color-1",
          },
          Array [
            Object {
              "borderRadius": 8,
              "borderWidth": 2,
              "padding": 16,
            },
            Object {
              "borderColor": "#C5CEE0",
            },
          ],
        ]
      }
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            undefined,
          ]
        }
        testID="bbcode-quote-text"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              undefined,
            ]
          }
          testID="bbcode-plain"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
          hello, world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`works with quote bbcode, name in quotes 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      appearance="default"
      level="1"
      style={
        Array [
          Object {
            "backgroundColor": "background-basic-color-1",
          },
          Array [
            Object {
              "borderRadius": 8,
              "borderWidth": 2,
              "padding": 16,
            },
            Object {
              "borderColor": "#C5CEE0",
            },
          ],
        ]
      }
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            Object {
              "fontStyle": "italic",
            },
          ]
        }
        testID="bbcode-quote-name"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        Foo
         wrote:
      </Text>
      <Text
        appearance="default"
        category="p1"
        style={
          Array [
            Object {
              "color": "#ff2",
              "fontFamily": "System",
              "fontSize": 15,
              "fontWeight": "400",
              "lineHeight": 20,
            },
            undefined,
          ]
        }
        testID="bbcode-quote-text"
        theme={
          Object {
            "text-basic-color": "#ff2",
            "text-primary-color": "#fff",
          }
        }
      >
        <Text
          appearance="default"
          category="p1"
          style={
            Array [
              Object {
                "color": "#ff2",
                "fontFamily": "System",
                "fontSize": 15,
                "fontWeight": "400",
                "lineHeight": 20,
              },
              undefined,
            ]
          }
          testID="bbcode-plain"
          theme={
            Object {
              "text-basic-color": "#ff2",
              "text-primary-color": "#fff",
            }
          }
        >
          hello, world
        </Text>
      </Text>
    </View>
  </View>
</View>
`;

exports[`works with url and link text bbcode 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello 
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      DUDE
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
       wassup
    </Text>
  </View>
</View>
`;

exports[`works with url auto matcher 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      https://dude.com
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      wassup
    </Text>
  </View>
</View>
`;

exports[`works with url auto matcher, query params 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      https://dude.com/?foo=bar
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      wassup
    </Text>
  </View>
</View>
`;

exports[`works with url auto matcher, trailing slash 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      https://dude.com/
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      wassup
    </Text>
  </View>
</View>
`;

exports[`works with url bbcode 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello 
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      https://dude.com
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
       wassup
    </Text>
  </View>
</View>
`;

exports[`works with url in quotes and link text bbcode 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello 
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      DUDE
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
       wassup
    </Text>
  </View>
</View>
`;

exports[`works with youtube auto replace (long domain) 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello
    </Text>
    <Text
      appearance="default"
      category="p1"
      status="primary"
      style={
        Array [
          Object {
            "color": "#fff",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-url"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      https://youtube.com/watch?v=123
    </Text>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      wassup
    </Text>
  </View>
</View>
`;

exports[`works with youtube auto replace (short domain) 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello
    </Text>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        allowsInlineMediaPlayback={true}
        cacheEnabled={true}
        domStorageEnabled={true}
        injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
        injectedJavaScriptForMainFrameOnly={true}
        javaScriptEnabled={true}
        messagingEnabled={false}
        source={
          Object {
            "uri": "https://www.youtube.com/embed/123 wassup?playsinline=1&fs=1",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "flex": 1,
            },
          ]
        }
        testID="bbcode-youtube"
        useSharedProcessPool={true}
      />
    </View>
  </View>
</View>
`;

exports[`works with youtube bbcode 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
      Hello 
    </Text>
    <View
      style={
        Array [
          Object {
            "flex": 1,
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <RNCWebView
        allowsInlineMediaPlayback={true}
        cacheEnabled={true}
        domStorageEnabled={true}
        injectedJavaScriptBeforeContentLoadedForMainFrameOnly={true}
        injectedJavaScriptForMainFrameOnly={true}
        javaScriptEnabled={true}
        messagingEnabled={false}
        source={
          Object {
            "uri": "https://www.youtube.com/embed/123?playsinline=1&fs=1",
          }
        }
        style={
          Array [
            Object {
              "flex": 1,
              "overflow": "hidden",
            },
            Object {
              "backgroundColor": "#ffffff",
            },
            Object {
              "flex": 1,
            },
          ]
        }
        testID="bbcode-youtube"
        useSharedProcessPool={true}
      />
    </View>
    <Text
      appearance="default"
      category="p1"
      style={
        Array [
          Object {
            "color": "#ff2",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
            "lineHeight": 20,
          },
          undefined,
        ]
      }
      testID="bbcode-plain"
      theme={
        Object {
          "text-basic-color": "#ff2",
          "text-primary-color": "#fff",
        }
      }
    >
       wassup
    </Text>
  </View>
</View>
`;
